python/tests/test_account.py

```diff
@@ -2446,25 +2446,24 @@ def test_immediate_autodelete(self, acfactory, lp):
 
         acfactory.wait_configure_and_start_io()
 
-        imap2 = ac2.direct_imap
-        imap2.idle_start()
-
         lp.sec("ac1: create chat with ac2")
         chat1 = ac1.create_chat(ac2)
         ac2.create_chat(ac1)
 
+        lp.sec("ac1: send message to ac2")
         sent_msg = chat1.send_text("hello")
-        imap2.idle_wait_for_new_message(terminate=False)
 
         msg = ac2._evtracker.wait_next_incoming_message()
         assert msg.text == "hello"
+        assert len(ac2.direct_imap.get_all_messages()) > 0
```

Comment:
this is racy -- core might be fast enough to delete the incoming message on the server and then direct_imap will not see any message and the assert will fail.

Reply:
fixed in forced push